prow/cmd/genjobs/genjobs.go

```diff
@@ -0,0 +1,382 @@
+func renameFile(pat string, src string, repl string) string {
+	return regexp.MustCompile(pat).ReplaceAllString(src, repl)
+}
+
+// convertOrgRepoStr translates the provided job org and repo based on the specified org mapping.
+func convertOrgRepoStr(s string) string {
+	a := strings.Split(s, "/")
+	org, repo := a[0], a[1]
+
+	valid := validateOrgRepo(org, repo)
+
+	if !valid {
+		return ""
+	}
+
+	return strings.Join([]string{orgMap[org], repo}, "/")
+}
+
+// updateUtilityConfig updates the jobs UtilityConfig fields based on provided inputs to work with private repositories.
+func updateUtilityConfig(job *config.UtilityConfig) {
+	if job.DecorationConfig == nil {
+		job.DecorationConfig = &prowjob.DecorationConfig{
+			GCSConfiguration: &prowjob.GCSConfiguration{
+				Bucket: bucket,
+			},
+			SSHKeySecrets: []string{sshKeySecret},
+		}
+	} else {
+		if job.DecorationConfig.GCSConfiguration == nil {
+			job.DecorationConfig.GCSConfiguration = &prowjob.GCSConfiguration{
+				Bucket: bucket,
+			}
+		} else {
+			job.DecorationConfig.GCSConfiguration.Bucket = bucket
+		}
+		if job.DecorationConfig.SSHKeySecrets == nil {
+			job.DecorationConfig.SSHKeySecrets = []string{sshKeySecret}
+		} else {
+			job.DecorationConfig.SSHKeySecrets = append(job.DecorationConfig.SSHKeySecrets, sshKeySecret)
+		}
+	}
+}
+
+// updateJobBase updates the jobs JobBase fields based on provided inputs to work with private repositories.
+func updateJobBase(job *config.JobBase, orgrepo string) {
+	job.Name = job.Name + jobnameSeparator + modifier
+	job.Cluster = cluster
+	job.CloneURI = fmt.Sprintf("git@github.com:%s.git", orgrepo)
+	job.Annotations = nil
+
+	if job.Labels == nil {
+		job.Labels = make(map[string]string)
+	}
+
+	for labelK, labelV := range labels {
+		job.Labels[labelK] = labelV
+	}
+
+}
+
+// getOutPath derives the output path from the specified input directory and current path.
+func getOutPath(p string, in string) string {
+	segments := strings.FieldsFunc(strings.TrimPrefix(p, in), func(c rune) bool { return c == '/' })
+
+	var org string
+	var repo string
+	var file string
+
+	if len(segments) >= 3 {
+		org = segments[len(segments)-3]
+		repo = segments[len(segments)-2]
+		file = segments[len(segments)-1]
+
+		if newOrg, ok := orgMap[org]; ok {
+			return filepath.Join(output, newOrg, repo, renameFile(`\b`+org+`\b`, file, newOrg))
+		}
+
+	} else if len(segments) == 2 {
+		org = segments[len(segments)-2]
+		file = segments[len(segments)-1]
+
+		if newOrg, ok := orgMap[org]; ok {
+			return filepath.Join(output, newOrg, renameFile(`\b`+org+`\b`, file, newOrg))
+		}
+
+	} else if len(segments) == 1 {
+		file = segments[len(segments)-1]
+
+		if !strings.HasPrefix(file, modifier) {
+			return filepath.Join(output, modifier+filenameSeparator+file)
+		}
+
+	}
+
+	return ""
+}
+
+// cleanOutPath deletes all files as the specified output path.
+func cleanOutPath(p string) {
+	for _, org := range orgMap {
+		_ = os.RemoveAll(filepath.Join(p, org))
```

Comment:
Shouldn't this output something if cleaning fails?

Reply:
If cleaning fails, conflicting files will still be overwritten, so it is not an error but I agree that it is a good idea to output a warning message to stderr to user.

Reply:
Added printing to stderr in appropriate places 👍